src/ode_solver/ode_solver_base.cpp

```diff
@@ -203,8 +203,10 @@ int ODESolverBase<dim,real,MeshType>::advance_solution_time (double time_advance
 {
     Parameters::ODESolverParam ode_param = ODESolverBase<dim,real,MeshType>::all_parameters->ode_solver_param;
 
-    const unsigned int number_of_time_steps = static_cast<int>(ceil(time_advance/ode_param.initial_time_step));
-    const double constant_time_step = time_advance/number_of_time_steps;
+   // const unsigned int number_of_time_steps = static_cast<int>(ceil(time_advance/ode_param.initial_time_step));
+    const unsigned int number_of_time_steps = (!this->all_parameters->use_energy) ? static_cast<int>(ceil(time_advance/ode_param.initial_time_step)) : this->current_iteration+1;
```

Comment:
what is going on here?

Reply:
For the energy tests because of the way the timestep is advanced in the unsteady testcases.

Reply:
fixed as per my explanation above.